### PR TITLE
Add initial presentation

### PR DESCRIPTION
**Pull request by katcipis**

### CODE REVIEW

File: gooo/gooo.slide [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+* What if...
+
+Object Oriented has been perfected ? And that is why later it
+developed inheritance ?
+
+
+* Evaluating both ideas
+
+The only way to understand if it object orientation
```
Comment (vitorarins) [inline gooo/gooo.slide]: typo: ~it~

File: gooo/gooo.slide [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+- Encapsulation (objects are like cells)
+- Messages between objects
+- No inheritance
+- No polymorphism (it is a subset of the idea)
+- Focus on what is between objects, not the form of the objects (morphos/type)
+
+
+* First Lesson Learned
+
+Giving names ir hard, even for people like Alan Kay =)
```
Comment (i4ki) [inline gooo/gooo.slide]: is*

File: gooo/gooo.slide [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+* What Go enables
+
+.image ./img/iointeraction-go-mul-interfaces.png _ 900
+
+
+* Why is this great ?
+
+- No need for the object implementing an interface to explicit declare it
+- The object implementing an interface is also not coupled to which name has been given to a interface (doesn't matter)
```
Comment (kamilash) [inline gooo/gooo.slide]: you use 'an interface' and latter 'a interface'.

File: gooo/gooo.slide [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+	syscall % grep -R ExitStatus .
+	./syscall_nacl.go:func (w WaitStatus) ExitStatus() int    { return 0 }
+	./syscall_bsd.go:func (w WaitStatus) ExitStatus() int {
+	./syscall_solaris.go:func (w WaitStatus) ExitStatus() int {
+	./syscall_linux.go:func (w WaitStatus) ExitStatus() int {
+	./syscall_windows.go:func (w WaitStatus) ExitStatus() int { return int(w.ExitCode) }
+	./syscall_plan9.go:func (w Waitmsg) ExitStatus() int { 
+
+
+* Its Something
```
Comment (kamilash) [inline gooo/gooo.slide]: It's

**Review by kamilash — COMMENTED** [review]
Looking forward to see it :D

File: gooo/gooo.slide (hunk 3 above)
Comment (katcipis) [reply gooo/gooo.slide]: which one is right ? =/

File: gooo/gooo.slide (hunk 4 above)
Comment (katcipis) [reply gooo/gooo.slide]: Im lazy when it comes to reaching out for ' =P

File: mkfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+present:V:
+	present .
+
+deps:V:
+	go install golang.org/x/tools/cmd/present
```
Comment (iuryfukuda) [inline mkfile]: maybe one `go get -v golang.org/x/tools/cmd/present` in deps too

Comment (zbioe) [inline mkfile:5]: maybe makes sense `go get -v golang.org/x/tools/cmd/present` in deps =s?

Comment (zbioe) [reply mkfile:5]: very nice this tool, did not know before

File: gooo/gooo.slide [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+@tiagokatcipis
+
+
+* Origins of Object Orientation
+
+- Every "new" thing usually has more than one influence
+- It is hard to pinpoint the origin of an idea
+- But people insist on being pedantic =D
+- Ranting like "Go is not OO"
+- So lets be pedantic too
```
Comment (zbioe) [inline gooo/gooo.slide]: let's

File: gooo/gooo.slide [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+
+* Alan Kay on Object Orientation
+
+I'm sorry that I long ago coined the term "objects" for this topic because
+it gets many people to focus on the lesser idea. The *big* *idea* is *"messaging"*
+
+The Japanese have a small word -- ma -- for "that which
+is in between" -- perhaps the nearest English equivalent is "interstitial"
+
+The key in making great and growable systems is much more to design how its
```
Comment (zbioe) [inline gooo/gooo.slide:73]: maybe `The key to making`

File: gooo/gooo.slide [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+
+* How to get the status code of a exec.Command ?
+
+
+* Seems trivial...
+
+- Actually not
+- Some operational systems uses integers
+- Others have strings as the results of a process execution
+- Officially exported API on exec.Command don't give access to the status code (only if it failed or not)
```
Comment (zbioe) [inline gooo/gooo.slide]: maybe `exec.Command doesn't give access`

File: gooo/gooo.slide [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+.code ./code/exitresult.go
+
+
+* Final Solution
+
+- This interface was not foreseen by the implementors of the WaitStatus
+- Yet it is there (for almost all platforms)
+- You can easily leverage that without changing the third party code (in this case the stdlib)
+- Sadly on this case the choice of exporting a *interface{}* obligates us to rely on runtime errors =(
+- Yet even with runtime cast we can see that we are not checking the object type, but its ability to satisty an interface
```
Comment (zbioe) [inline gooo/gooo.slide]: `satisfy` ?

**Comment by zbioe** [comment]
niceeee presentatioon, i'm looking forward to it

File: gooo/gooo.slide (hunk 7 above)
Comment (katcipis) [reply gooo/gooo.slide:73]: This is a quote from alan kay dude...do you want to correct ALAN KAY  ? =P

File: mkfile (hunk 5 above)
Comment (katcipis) [reply mkfile:5]: indeed it is =D

I'm always in doubt if go install will also already go get the code or if it just install if the code has already been got =P. Just running the go install failed on your host ?

Comment (zbioe) [reply mkfile:5]: I have, just had to prefix the go get before, but I do not know, sometimes it was a bad configuration, the bin of the path was not configured correctly too

Comment (zbioe) [reply mkfile:5]: I have, just had to prefix the go get before, but I do not know, sometimes it was a bad configuration, the bin of the path was not configured correctly too

Comment (zbioe) [reply mkfile:5]: I have, just had to prefix the go get before, but I do not know, sometimes it was a bad configuration, the bin of the path was not configured correctly too

Comment (zbioe) [reply mkfile:5]: I have, just had to prefix the go get before, but I do not know, sometimes it was a bad configuration, the bin of the path was not configured correctly too

File: gooo/gooo.slide (hunk 7 above)
Comment (zbioe) [reply gooo/gooo.slide:73]: hdudhhhdu, my bad, sorry for that

**Pull request merged**